Modix/Modules/MoveMessageModule.cs

```diff
@@ -0,0 +1,102 @@
+                    Context.User.Mention, messageId);
+            }
+
+            // Format output
+            var builder = new EmbedBuilder()
+                    .WithColor(new Color(95, 186, 125))
+                    .WithDescription(message.Content);
+
+            if (!string.IsNullOrWhiteSpace(reason))
+            {
+                builder.AddInlineField("Reason", reason);
+            }
+
+            var mover = $"@{(Context.User as SocketGuildUser)?.Nickname ?? Context.User.Username}";
+            builder.WithFooter($"Message copied from #{message.Channel.Name} by {mover}");
+
+            await channel.SendMessageAsync("", embed: builder);
+
+            // Delete the source message, and the command message that started this request
+            await message.DeleteAsync();
+            await Context.Message.DeleteAsync();
+            await Context.Channel.SendMessageAsync($"Message {messageId} moved by {mover}");
+        }
+
+        [Command("move"), Summary("Moves a message from one channel to another."), Remarks("Usage: !move 12345 #foo")]
+        public async Task Run(ulong messageId, SocketTextChannel channel)
+            => await Run(messageId, channel, null);
+
+        [Command("move"), Summary("Moves a message from one channel to another."), Remarks("Usage: !move 12345 #foo")]
+        public async Task Run(SocketTextChannel channel, ulong messageId)
+            => await Run(messageId, channel, null);
+
+        [Command("move"), Summary("Moves a message from one channel to another."), Remarks("Usage: !move 12345 #foo")]
+        public async Task Run(SocketTextChannel channel, ulong messageId, [Remainder] string reason)
+            => await Run(messageId, channel, null);
+
+        private async Task<IMessage> FindMessageInUnknownChannel(ulong messageId)
+        {
+            IMessage message = null;
+
+            // We haven't found a message, now fetch all text
+            // channels and attempt to find the message
+
+            var channels = await Context.Guild.GetTextChannelsAsync();
+
+            foreach (var channel in channels)
+            {
+                try
+                {
+                    message = await GetMessage(messageId, channel);
+
+                    if (message != null)
+                        break;
+                }
+                catch (Exception e)
+                {
+                    Log.Debug(e, "Failed accessing channel {ChannelName} when searching for message {MessageId}",
+                        channel.Name, messageId);
+                }
+            }
+
+            return message;
+        }
+
+        private Task<IMessage> GetMessage(ulong messageId, ITextChannel channel)
+            => GetMessageInChannel(messageId, channel);
+
+        private static Task<IMessage> GetMessageInChannel(ulong messageId, ITextChannel channel)
+            => channel.GetMessageAsync(messageId);
+    }
```

Comment:
These methods don't do anything except inflate the line count. Remove in favor of the method on ITextChannel.